src/pages/repository/content.js

```diff
@@ -28,9 +51,95 @@ class Content extends React.Component {
     }
   }
 
+  componentDidUpdate() {
+    const {builds, state} = this.props;
+
+    // populating build filter tag suggestions
+    if (builds && (state == undefined || state.suggestions == undefined || state.suggestions.length == 0)) {
+      this.setFilterSuggestions();
+    }
+  }
+
+  setFilterSuggestions() {
+    const {owner, name} = this.props.params;
+    const {builds} = this.props;
+
+    var dict = {};
+    var item;
+
+    Object.keys(builds).forEach(k => {
+      item = builds[k];
+      dict['author:' + item.author] = true;
+      dict['branch:' + item.branch] = true;
+      dict['status:' + item.status] = true;
+      dict['event:' + item.event] = true;
+      if(item.ref.includes('tags')) {
+        dict['tag:' + item.ref.replace('refs/tags/', '')] = true;
+      }
+      if(item.deploy_to !== '') {
+        dict['deploy_to:' + item.deploy_to] = true;
+      }
+    });
+
+    var suggestions = Object.keys(dict).sort((a, b) => {return a > b;});
+    suggestions = suggestions.map(item => {return {label:item, value:item};});
+    events.emit(BUILD_FILTER_SUGGESTIONS, {owner, name, suggestions});
+  }
```

Comment:
I would rewrite this mutating style code like so:

```js
const terms = Object.keys(build).map(k => builds[k]).reduce((dict, build) => {
  dict['author: ' + build.author] = true;
  return dict;
}, {});
```

Also please replace the `var`s with `const`s:

```js
const suggestions = Object.keys(terms)
  .sort((a, b) => a > b)
  .map(item => ({label:item, value:item}));
```